src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java

```diff
@@ -123,6 +130,14 @@ public AnalysisManager(LanguageClientLogOutput lsLogOutput, EnginesFactory engin
     this.bindingManager = bindingManager;
     this.analysisExecutor = Executors.newSingleThreadExecutor(Utils.threadFactory("SonarLint analysis", false));
     this.watcher = new EventWatcher();
+    this.taintVulnerabilitiesPerFile = new ConcurrentHashMap<>();
+  }
+
+  public AnalysisManager(LanguageClientLogOutput lsLogOutput, EnginesFactory enginesFactory, SonarLintExtendedLanguageClient client, SonarLintTelemetry telemetry,
+    WorkspaceFoldersManager workspaceFoldersManager, SettingsManager settingsManager, ProjectBindingManager bindingManager,
+    Map<URI, List<ServerIssue>> taintVulnerabilitiesPerFile) {
+    this(lsLogOutput, enginesFactory, client, telemetry, workspaceFoldersManager, settingsManager, bindingManager);
+    this.taintVulnerabilitiesPerFile = taintVulnerabilitiesPerFile;
```

Comment:
I don't like this pattern. It may lead to forget to initialize the attribute if another constructor is added later. IMO it is better to have:
```
final fieldA;

// For testing
constructor(fieldA) {
  this.fieldA = fieldA;
}

constructor() {
  this(defaultValueForFieldA);
}

```